### PR TITLE
added currency field in exp claim and emp adv

### PR DESCRIPTION
**Pull request by shubhamgsuite42**

### CODE REVIEW

File: hrms/hr/doctype/employee_advance/employee_advance.js [hunk 1]
```diff
@@ -123,7 +127,7 @@ frappe.ui.form.on('Employee Advance', {
 			});
 		});
 
-		if(frm.doc.status === "Pending Payment" && frm.doc.currency == "INR"){
```
Comment (dinakarreddy) [inline hrms/hr/doctype/employee_advance/employee_advance.js:126]: check with company

File: hrms/hr/doctype/expense_claim/expense_claim.js [hunk 2]
```diff
@@ -208,6 +208,14 @@ frappe.ui.form.on("Expense Claim", {
 		});
 	},
 
+	currency: function(frm) {
+		if(frm.doc.currency == "INR"){
```
Comment (dinakarreddy) [inline hrms/hr/doctype/expense_claim/expense_claim.js]: check with company currency != usr currency

File: hrms/overrides/custom_employee_advance.py [hunk 3]
```diff
@@ -53,6 +53,7 @@ def validate(self):
         ]
         self.add_approver()
         self.validate_approver()
+        self.calulate_inr_amount()
```
Comment (dinakarreddy) [inline hrms/overrides/custom_employee_advance.py]: rename

File: hrms/overrides/custom_employee_advance.py [hunk 4]
```diff
@@ -275,6 +276,11 @@ def validate_approver(self):
         ):
             frappe.throw(_(f"Approval selected does not have L1 or L2 Expense Approver Role"))
 
+    def calulate_inr_amount(self):
+        get_default_company_currency = frappe.db.get_value("Company", CompanyConstants.SUITE42, "default_currency")
+        if self.currency != get_default_company_currency:
+            self.advance_amt_inr = flt(self.exchange_rate*self.advance_amount, 1)
```
Comment (dinakarreddy) [inline hrms/overrides/custom_employee_advance.py]: make exchange rate readonly and ask for amount in user currency and amount in company currency

File: hrms/overrides/custom_expense_claim.py [hunk 5]
```diff
@@ -93,6 +94,17 @@ def validate_mmit_id(self):
                     _(f"MMT Should not be attached for Expense Category {self.expense_category}")
                 )
 
+    def calulate_inr_amount(self):
```
Comment (dinakarreddy) [inline hrms/overrides/custom_expense_claim.py]: rename to validate expense amounts

File: hrms/hr/doctype/employee_advance/employee_advance.js (hunk 1 above)
Comment (shubhamgsuite42) [reply hrms/hr/doctype/employee_advance/employee_advance.js:126]: Done, comparing with company currency wherever required. For showing pay button not required to compare company currency.

File: hrms/overrides/custom_employee_advance.py (hunk 3 above)
Comment (shubhamgsuite42) [reply hrms/overrides/custom_employee_advance.py]: Done.

File: hrms/overrides/custom_employee_advance.py (hunk 4 above)
Comment (shubhamgsuite42) [reply hrms/overrides/custom_employee_advance.py]: Done.

File: hrms/overrides/custom_expense_claim.py (hunk 5 above)
Comment (shubhamgsuite42) [reply hrms/overrides/custom_expense_claim.py]: Done

File: hrms/hr/doctype/expense_claim/expense_claim.js (hunk 2 above)
Comment (shubhamgsuite42) [reply hrms/hr/doctype/expense_claim/expense_claim.js]: Done

**Pull request merged**